### PR TITLE
- add meson

### PR DESCRIPTION
**Pull request by SSE4**
Specify library name and version:  **meson/0.51.2**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

**Comment by SSE4** [comment]
depends on https://github.com/conan-io/conan-center-index/pull/125

File: recipes/meson/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+from conans import ConanFile, tools
+import os
+
+
+class MesonInstallerConan(ConanFile):
+    name = "meson"
+    version = "0.51.2"
+    description = "Meson is a project to create the best possible next-generation build system"
+    topics = ("conan", "meson", "mesonbuild", "build-system")
+    url = "https://github.com/bincrafters/conan-meson_installer"
```
Comment (Croydon) [inline recipes/meson/all/conanfile.py]:
```suggestion
    url = "https://github.com/conan-io/conan-center-index"
```

File: recipes/meson/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    homepage = "https://github.com/mesonbuild/meson"
+    author = "Bincrafters <bincrafters@gmail.com>"
+    license = "Apache-2.0"
+    no_copy_source = True
+    exports = ["LICENSE.md"]
```
Comment (Croydon) [inline recipes/meson/all/conanfile.py]:
```suggestion
    exports = ["LICENSE.md"]
```

File: recipes/meson/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+
+
+class MesonInstallerConan(ConanFile):
+    name = "meson"
+    version = "0.51.2"
+    description = "Meson is a project to create the best possible next-generation build system"
+    topics = ("conan", "meson", "mesonbuild", "build-system")
+    url = "https://github.com/bincrafters/conan-meson_installer"
+    homepage = "https://github.com/mesonbuild/meson"
+    author = "Bincrafters <bincrafters@gmail.com>"
```
Comment (Croydon) [inline recipes/meson/all/conanfile.py]:
```suggestion
```

**Pull request merged**